sdks/java/io/common/src/test/java/org/apache/beam/sdk/io/common/SchemaAwareJavaBeans.java

```diff
@@ -270,6 +400,55 @@ public static SerializableFunction<Row, DoublyNestedDataTypes> doublyNestedDataT
     return DEFAULT_SCHEMA_PROVIDER.fromRowFunction(DOUBLY_NESTED_DATA_TYPES_TYPE_DESCRIPTOR);
   }
 
+  /**
+   * Contains all primitive Java types supported by Avro. The purpose of this class is to test
+   * schema-aware PTransforms with flat {@link Schema} {@link Row}s.
+   */
+  @DefaultSchema(AutoValueSchema.class)
+  @AutoValue
+  public abstract static class AvroPrimitiveDataTypes implements Serializable {
```

Comment:
I'll let you decide based on my comment and I'll LGTM either way.

What I like about AvroPrimitiveDataTypes is that it makes it easy to write tests against field types we know would be compatible with Avro encodings.  What I struggle with is:
1. The name implies a coupling of data encoding though I get it
2. Writing tests against AllPrimitiveDataTypes that fail is valuable information on how we expose FieldType
3. Dealing with annoying FieldTypes that aren't compatible with a Schema related use case is valuable information to figure out how a real user of Beam would have to deal with this when they don't have the ability to change the original class but do have the ability to change the Schema using, for example, https://beam.apache.org/releases/javadoc/current/org/apache/beam/sdk/schemas/transforms/DropFields.html.  I dealt with this problem when writing tests for CsvIO and it was valuable discovery.  Writing tests around this problem of incompatible fields provided for me a usability test not just does this code work test.

Another alternative would be just to get rid of the Byte field in AllPrimitiveDataTypes but that would require you to refactor the tests that break as a result.

This comment also applies to the other additions to SchemaAwareJavaBeans and I didn't comment on those individually as it would have been redundant.

Reply:
Yeah the problem is Short (int16)  and Byte (a single Byte) are not types that are supported by Avro (see [Avro primitive types](https://docs.oracle.com/cd/E26161_02/html/GettingStartedGuide/avroschemas.html#:~:text=in%20this%20chapter.-,Primitive%20Data%20Types,-In%20the%20previous)) or Parquet (see [primitive types](https://parquet.apache.org/docs/file-format/types/)). So actually in AvroUtils we turn Short and Byte types into Beam Schema types to an Integer (int32).

Avro and Parquet do support a sequence of bytes though, so I've replaced Byte with ByteBuffer in `AvroPrimitiveDataTypes`.

>Another alternative would be just to get rid of the Byte field in AllPrimitiveDataTypes

If you agree, I'd be happy to remove Short and replace Byte with ByteBuffer in `AllPrimitiveDataTypes` and use that for everything instead (for both read and write).

Reply:
If you need my help refactoring tests that break as a result of removing Short and Byte, please let me know and I can commit to this PR.

Reply:
Tried this, looks like XML, JSON, CSV don't support bytes (as in a sequence of bytes), but do support a single byte type. Opposite to the case of Avro and Parquet, which support a byte sequence but not a single byte.

Reply:
If you're not opposed, will proceed by not including any form of bytes in `AllPrimitiveDataTypes` since there is no common overlap. Instead, we can have two new classes, ByteType and ByteSequenceType that will test for byte, byte[]. I can create new tests for these in both file read and write transforms.

Reply:
I like your idea about removing the troublesome byte types out.  Let me know if you need help refactoring tests.